src/ipfs/daemon.rs

```diff
@@ -0,0 +1,221 @@
+                println!("Signal {} received. Attempting to stop IPFS...", sig);
+                match me.shutdown() {
+                    Ok(_) => println!("{}", "IPFS has shutdown successfully.".green()),
+                    Err(_) => println!("{}", "IPFS failed to shutdown succefully! You may need to stop the proccess yourself.".red())
+                };
+            });
+        });
+
+        println!("{}", "IPFS has launched successfully!!".green());
+        Ok(())
+    }
+
+    /// This method sends an http signal to the IPFS deamon to shutdown. 
+    /// 
+    /// Note 1: This method should work even if the daemon was not started by this proccess.
+    /// 
+    /// Note 2: This method return when it recieves an http response from the daemon, and not
+    /// when the proccess has actually stopped.
+    pub fn shutdown(&self) -> Result<()>{
+        self.tokio.block_on(async {
+            block_on(self.client.shutdown())
+        })?;
+        Ok(())
+    }
+
+    async fn is_ipfs_ready(&self) -> bool {
+        let res = self.tokio.block_on(async {
+            self.client.config_show().await
+        });
+        // if res.is_ok() {
+        //     println!("Config is {}", res.as_ref().unwrap());
+        // }
+        return match res {
+            Ok(_) => true,
+            Err(_) => false
+        };
+    }
+
+    async fn await_ready(&self) -> Result<()> {
+        let start_time = SystemTime::now();
+        loop {
+            println!("{}", "checking if ipfs is ready...".green());
+
+            if self.is_ipfs_ready().await {
+                println!("{}", "IPFS is ready!!".green());
+                break;
+            }
+
+            thread::sleep(Duration::new(IPFS_SLEEP_LENGTH as u64, 0));
+
+            let now = SystemTime::now();
+            if now.duration_since(start_time)? > Duration::new(IPFS_BOOT_TIME_OUT as u64, 0) {
+                bail!(
+                    "{}",
+                    "Failed to start ipfs because the timeout reached!!".red()
+                )
+            }
+        }
+        anyhow::Ok(())
+    }
+}
+
+impl TryFrom<IpfsConnInfo> for IpfsDaemon{
+    type Error = anyhow::Error;
+    /// This is one of the two main ways to make a new instance of the IPFS daemon struct. Use this
+    /// method when you want to make a new instance and you need to set the port and address of
+    /// the daemon you are attempting to refrence
+    fn try_from(conn_info: IpfsConnInfo) -> Result<Self> {
+        let client = IpfsClient::from_host_and_port(
+            "http".parse()?, 
+            &conn_info.address, 
+            conn_info.port
+        )?;
+
+        let runtime = tokio::runtime::Runtime::new()?;
+        Ok(Self{
+            client,
+            tokio: runtime,
+            conn_info
+        })   
+    }
+}
+
+impl Default for IpfsDaemon {
+    /// This is one of the two main ways to make a new instance of the IPFS daemon struct. Use this
```

Comment:
Are there other ways to make a new instance aside for these two ways? If not, it would be alright drop "main" from the comments. It makes me think there might be other ways to create an instance.